<launch>
  <!-- <arg name="input_image" default="/head_camera/rgb/image_raw" /> -->
  <arg name="input_image" default="/head_camera_remote/rgb/image_rect_color" />
  <arg name="input_depth_image" default="/head_camera_remote/depth/image_rect_raw" />
  <arg name="input_camera_info" default="/head_camera/depth_registered/camera_info" />

  <arg name="box_len_x" default="0.14" />
  <arg name="box_len_y" default="0.2" />
  <arg name="box_len_z" default="0.12" />

  <node name="paper_finder"
        pkg="jsk_perception" type="paper_finder.py"
        output="screen" >
    <remap from="~input" to="$(arg input_image)" />
    <remap from="~input/depth" to="$(arg input_depth_image)" />
    <remap from="~input/camera_info" to="$(arg input_camera_info)" />
    <rosparam>
      area_tolerance: 0.1
      approximate_sync: true
      length_tolerance: 0.2
      rect_x: 0.3
      rect_y: 0.6
      width_depth: true
    </rosparam>
  </node>

  <!-- <!-\- paper edge -\-> -->
  <!-- <node name="edge_finder_paper" -->
  <!--       pkg="jsk_perception" type="edge_finder.py" -->
  <!--       output="screen" > -->
  <!--   <remap from="~input" to="$(arg input_image)" /> -->
  <!--   <remap from="~input/depth" to="$(arg input_depth_image)" /> -->
  <!--   <remap from="~input/camera_info" to="$(arg input_camera_info)" /> -->
  <!--   <rosparam> -->
  <!--     approximate_sync: true -->
  <!--     length_tolerance: 0.1 -->
  <!--     <!-\- box -\-> -->
  <!--     <!-\- ps_list_a_x: 0.52 -\-> -->
  <!--     <!-\- ps_list_a_y: -0.035 -\-> -->
  <!--     <!-\- ps_list_a_z: 0.046 -\-> -->
  <!--     <!-\- ps_list_b_x: 0.38 -\-> -->
  <!--     <!-\- ps_list_b_y: -0.035 -\-> -->
  <!--     <!-\- ps_list_b_z: 0.046 -\-> -->
  <!--     <!-\- cylinder -\-> -->
  <!--     ps_list_a_x: 0.52 -->
  <!--     ps_list_a_y: 0.05 -->
  <!--     ps_list_a_z: 0.1 -->
  <!--     ps_list_b_x: 0.38 -->
  <!--     ps_list_b_y: 0.05 -->
  <!--     ps_list_b_z: 0.1 -->
  <!--     source_frame: "/head_camera_rgb_optical_frame" -->
  <!--     target_frame: "/WAIST" -->
  <!--     width_depth: true -->
  <!--   </rosparam> -->
  <!-- </node> -->
  <!-- <!-\- box edge -\-> -->
  <!-- <node name="edge_finder_box" -->
  <!--       pkg="jsk_perception" type="edge_finder.py" -->
  <!--       output="screen" > -->
  <!--   <remap from="~input" to="$(arg input_image)" /> -->
  <!--   <remap from="~input/depth" to="$(arg input_depth_image)" /> -->
  <!--   <remap from="~input/camera_info" to="$(arg input_camera_info)" /> -->
  <!--   <rosparam> -->
  <!--     approximate_sync: true -->
  <!--     length_tolerance: 0.1 -->
  <!--     ps_list_a_x: 0.52 -->
  <!--     ps_list_a_y: 0.065 -->
  <!--     ps_list_a_z: 0.046 -->
  <!--     ps_list_b_x: 0.38 -->
  <!--     ps_list_b_y: 0.065 -->
  <!--     ps_list_b_z: 0.046 -->
  <!--     box_len_x: 0.14 -->
  <!--     target_frame: "/WAIST" -->
  <!--     width_depth: true -->
  <!--   </rosparam> -->
  <!-- </node> -->

  <group ns="paper_finder" >
    <node name="republish"
          pkg="image_transport" type="republish"
          args="raw compressed">
      <remap from="in" to="output/viz" />
      <remap from="out" to="output/viz" />
    </node>
  </group>

  <!-- <group ns="edge_finder_paper" > -->
  <!--   <node name="republish" -->
  <!--         pkg="image_transport" type="republish" -->
  <!--         args="raw compressed"> -->
  <!--     <remap from="in" to="output/viz" /> -->
  <!--     <remap from="out" to="output/viz" /> -->
  <!--   </node> -->
  <!-- </group> -->

  <!-- <group ns="edge_finder_box" > -->
  <!--   <node name="republish" -->
  <!--         pkg="image_transport" type="republish" -->
  <!--         args="raw compressed"> -->
  <!--     <remap from="in" to="output/viz" /> -->
  <!--     <remap from="out" to="output/viz" /> -->
  <!--   </node> -->
  <!-- </group> -->

  <!-- publish marker array -->
  <node name="box_marker_array_publisher" type="marker_array_pub.py" pkg="wrapping_melodic" output="screen" >
    <param name="~box_z_coord" value="0.04"/>
    <param name="~box_len_x" value="$(arg box_len_x)"/>
    <param name="~box_len_y" value="$(arg box_len_y)"/>
    <param name="~box_len_z" value="$(arg box_len_z)"/>
  </node>

</launch>
